src/analyze.jl

```diff
@@ -157,10 +159,91 @@ function _slic(alg::SLIC, img::AbstractArray{<:Lab, 2})
+        segment_size = height * width / n_segments
+        min_size = round(Int, 0.5 * segment_size)
+        max_size = round(Int, 3.0 * segment_size)
+
+        dy = [1, -1, 0, 0]
+        dx = [0, 0, 1, -1]
+        # dz = [] # reversed for supervoxels
+
+        start_label = 1
+        mask_label = start_label - 1 # indicates the label of this pixel has not been assigned
+        nearest_segments_final = fill(mask_label, height, width)
+        current_new_label = start_label
+
+        # used for BFS
+        current_segment_size = 1
+        bfs_visited = 0
+
+        # store neighboring pixels
+        # now set the dimension to 2 because we are using superpixel
+        coord_list = fill(0, max_size, 2)
+ 
+        for x = 1:width
+            for y = 1:height
+                if nearest_segments[y, x] == mask_label continue end
+                if nearest_segments_final[y, x] > mask_label continue end
+
+                adjacent = 0
+                label = nearest_segments[y, x]
+                nearest_segments_final[y, x] = current_new_label
+                current_segment_size = 1
+                bfs_visited = 0
+                coord_list[bfs_visited + 1, 1] = y
+                coord_list[bfs_visited + 1, 2] = x
+
+                # Preform BFS to find the size of superpixel with 
+                # same lable number
+                while bfs_visited < current_segment_size <= max_size
+                    for i = 1:4
+                        yy = coord_list[bfs_visited + 1, 1] + dy[i]
+                        xx = coord_list[bfs_visited + 1, 2] + dx[i]
+
+                        if 1 <= yy <= height &&  1 <= xx <= width
+                            if nearest_segments[yy, xx] == label && nearest_segments_final[yy, xx] == mask_label
+                                nearest_segments_final[yy, xx] = current_new_label
+                                coord_list[current_segment_size + 1, 1] = yy # <-- index problem in the future
+                                coord_list[current_segment_size + 1, 2] = xx # <-- index problem in the future
+                                current_segment_size += 1
+                                
+                                if current_segment_size > max_size break end
+                            elseif nearest_segments_final[yy, xx] > mask_label &&
+                                   nearest_segments_final[yy, xx] != current_new_label
+                                adjacent = nearest_segments_final[yy, xx]
+                            end
+                        end
+                    end
+                    bfs_visited += 1
+                end
+
+                #for i = 1:current_segment_size println(coord_list[i, 1]) end
+                #println(max_size, " ", current_segment_size)
+
+                # merge the superpixel to its neighbor if it is too small
+                if current_segment_size < min_size
+                    for i = 1:current_segment_size
```

Comment:
if there's no if statements in the for-loop, it's usually a good suggestion to add `@simd` macro to do the CPU level parallelization.

```suggestion
                    @inbounds @simd for i = 1:current_segment_size
```

also remove `println`.

Reply:
To be clear. This and the `@inbounds` comments are for minor performance tweaks. I'm not sure if these are useful, so benchmarks are needed. If you believe these are useless, feel free to ignore them.